internal/future/app/mmlogic/mmlogic_service.go

```diff
@@ -15,31 +15,139 @@
 package mmlogic
 
 import (
+	"time"
+
+	"encoding/json"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/sirupsen/logrus"
+	"open-match.dev/open-match/internal/config"
 	"open-match.dev/open-match/internal/future/pb"
+	"open-match.dev/open-match/internal/future/storage"
+	"open-match.dev/open-match/internal/set"
+
+	"fmt"
+)
+
+var (
+	logger = logrus.WithFields(logrus.Fields{
+		"app":       "openmatch",
+		"component": "app.mmlogic",
+	})
 )
 
 // The MMLogic API provides utility functions for common MMF functionality such
 // as retreiving Tickets from state storage.
 type mmlogicService struct {
+	storageService storage.Service
+	cfg            config.View
 }
 
 // RetrievePool gets the list of Tickets that match every Filter in the
 // specified Pool.
 // TODO: Consider renaming to "GetPool" to be consistent with HTTP REST CRUD
 // conventions. Right now there's a GET and a POST for this verb.
-func (s *mmlogicService) RetrievePool(req *pb.RetrievePoolRequest, stream pb.MmLogic_RetrievePoolServer) error {
-	ctx := stream.Context()
+func (s *mmlogicService) RetrievePool(req *pb.RetrievePoolRequest, responseServer pb.MmLogic_RetrievePoolServer) error {
+	poolName := req.Pool.Name
+	poolFilters := req.Pool.Filter
+	redisConn := storage.GetDeprecatedRedisPool(s.storageService).Get()
```

Comment:
I'd say we should stuff the redis read data into the client now. Let's not use redis directly here. @sawagh Submitted the API shell for `statestore`.